curiosrc/alertmanager/alerts.go

```diff
@@ -330,3 +334,120 @@ func (al *alerts) getAddresses() ([]string, []string, error) {
 	}
 	return uniqueAddrs, miners, nil
 }
+
+func wdPostCheck(al *alerts) {
+	Name := "WindowPost"
+
+	head, err := al.api.ChainHead(al.ctx)
+	if err != nil {
+		al.alertMap[Name].err = err
+		return
+	}
+
+	from := head.Height() - abi.ChainEpoch(math.Ceil(float64(AlertMangerInterval)/float64(build.BlockDelaySecs))) - 1
+
+	var wdDetails []struct {
+		Miner     int64          `db:"sp_id"`
+		Deadline  int64          `db:"deadline"`
+		Partition int64          `db:"partition"`
+		Epoch     abi.ChainEpoch `db:"submit_at_epoch"`
+		Proof     []byte         `db:"proof_params"`
+	}
+
+	err = al.db.Select(al.ctx, &wdDetails, `
+				SELECT sp_id, submit_at_epoch, proof_params, partition, deadline
+				FROM wdpost_proofs 
+				WHERE submit_at_epoch > $1;`, from)
+	if err != nil {
+		al.alertMap[Name].err = xerrors.Errorf("getting windowPost details from database: %w", err)
+		return
+	}
+
+	for _, detail := range wdDetails {
+		var postOut *miner.SubmitWindowedPoStParams
+		err = postOut.UnmarshalCBOR(bytes.NewReader(detail.Proof))
+		if err != nil {
+			al.alertMap[Name].err = xerrors.Errorf("unmarshaling windowPost proof params: %w", err)
+			return
+		}
+
+		for i := range postOut.Partitions {
+			c, err := postOut.Partitions[i].Skipped.Count()
+			if err != nil {
+				al.alertMap[Name].err = xerrors.Errorf("getting skipped sector count: %w", err)
+				return
+			}
+			if c > 0 {
+				al.alertMap[Name].alertString += fmt.Sprintf("Skipped %d sectors in deadline %d partition %d. ", c, postOut.Deadline, postOut.Partitions[i].Index)
+			}
+		}
+	}
+}
+
+func wnPostCheck(al *alerts) {
+	Name := "WinningPost"
+	head, err := al.api.ChainHead(al.ctx)
+	if err != nil {
+		al.alertMap[Name].err = err
+		return
+	}
+
+	from := head.Height() - abi.ChainEpoch(math.Ceil(float64(AlertMangerInterval)/float64(build.BlockDelaySecs))) - 1
+
+	var wnDetails []struct {
+		Miner int64          `db:"sp_id"`
+		Block string         `db:"mined_cid"`
+		Epoch abi.ChainEpoch `db:"epoch"`
+	}
+
+	err = al.db.Select(al.ctx, &wnDetails, `
```

Comment:
We should also count tasks without checking `won` and make sure we're seeing the correct count

Reply:
Fixed.